### PR TITLE
{phys}[gompi/2021b] QuickPIC v20210224

### PR DESCRIPTION
**Pull request by sassy-crick**
(created using `eb --new-pr`)

Requires:
- [x] PR #15979

### CODE REVIEW

File: easybuild/easyconfigs/q/QuickPIC/quickpic-makefile.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ FC_GF_OPENMPI = mpif90 -c -I$(JSON_INC) -fopenmp
+ CC_GF_OPENMPI = mpicc -c -O3 -std=c99
+@@ -14,11 +14,11 @@ FORMAT_FIXED_GF_OPENMPI = -ffixed-form
+ 
+ # hdf5 libraries 
+ 
+-HDF5_LIB =/usr/local/hdf5/lib
+-HDF5_INC =/usr/local/hdf5/include
++HDF5_LIB =${EBROOTHDF5}/include
++HDF5_INC =${EBROOTHDF5}/lib
```
Comment (bedroge) [inline easybuild/easyconfigs/q/QuickPIC/quickpic-makefile.patch:24]: @sassy-crick I think you mixed these up ;-)
```suggestion
+HDF5_LIB =${EBROOTHDF5}/lib
+HDF5_INC =${EBROOTHDF5}/include
```
Note that due to this change, the checksum will also have to be recalculated for the patch file.

**Review by bedroge — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/q/QuickPIC/quickpic-makefile.patch (hunk 1 above)
Comment (sassy-crick) [reply easybuild/easyconfigs/q/QuickPIC/quickpic-makefile.patch]: Yes, that was me correcting one thing and got it wrong in the other place. Thanks for finding that! All done now.

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
pg-interactive.hpc.rug.nl - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 3.6.8
See https://gist.github.com/2574ad45f0d5d11cb829854307794717 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.4
See https://gist.github.com/2930ffe468cdd8a4ad43c1e2d979892c for a full test report.

File: easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+sources = [{'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ}]
+patches = ['quickpic-makefile.patch']
+
+checksums = [
+    '83d3664542ce0a1c1fa055802e214bdba98790ee512f1f2525d0c087d20d88af',  # QuickPIC-20210224.tar.gz
+    'fd4be47519513ca3add974e5c5964707a7534ef68ad850f775fae3d8e0cdc987',  # quickpic-makefile.patch
+]
+
+dependencies = [
+    ('HDF5', '1.10.8'),
```
Comment (bedroge) [inline easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb:30]: The test suite is still complaining about the version of HDF5: the default for `gompi/2021b` is `1.12.1`, can we stick to that version, or is this older version really required?

Comment (sassy-crick) [reply easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb]: I managed to change that to the newer version, that seems to work here. I left a comment where that comes from in the EC file which is addressing the changes in the HDF5 API and how to fix that. Thanks for pointing that out.

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.4
See https://gist.github.com/94e7334e3d4baa8611d8fa96cb1ab91c for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
pg-interactive.hpc.rug.nl - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 3.6.8
See https://gist.github.com/2a39e4f10eca90e353349fff406d1b6a for a full test report.

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+which is developed based on the framework UPIC. QuickPIC can efficiently simulate plasma 
+based accelerator problems. This is the UCLA Plasma Simulation Group's official open-source 
+repository for QuickPIC."""
+
+toolchain = {'name': 'gompi', 'version': '2021b'}
+toolchainopts = {'pic': True, 'usempi': True, 'cstd': 'c99'}
+
+source_urls = ['https://github.com/UCLA-Plasma-Simulation-Group/QuickPIC-OpenSource/archive']
+sources = [{'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ}]
+patches = ['quickpic-makefile.patch']
```
Comment (bedroge) [inline easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb:22]: One last thing (had overlooked that) is the name of the patch file. `quickpic` should definitely have the same capitalization as `name`. Furthermore, I don't know if this patch is version-specific, but maybe it should be something like:
```suggestion
patches = ['QuickPIC-20210224_fix_paths_in_makefile.patch']
```
?

File: easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+checksums = [
+    '83d3664542ce0a1c1fa055802e214bdba98790ee512f1f2525d0c087d20d88af',  # QuickPIC-20210224.tar.gz
+    'fd4be47519513ca3add974e5c5964707a7534ef68ad850f775fae3d8e0cdc987',  # quickpic-makefile.patch
```
Comment (bedroge) [inline easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb:26]: The name should be changed here too

**Review by bedroge — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb (hunk 3 above)
Comment (bedroge) [reply easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb:22]: And the file itself still has to be renamed then as well, of course.

Comment (sassy-crick) [reply easybuild/easyconfigs/q/QuickPIC/QuickPIC-20210224-gompi-2021b.eb]: Done. I never really know how to name the files, so thanks for your suggestion!

**Comment by bedroge** [comment]
@boegelbot please test @ generoso

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.4
See https://gist.github.com/beb6c6a1be5477f863d41bc527045a04 for a full test report.

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
pg-interactive.hpc.rug.nl - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 3.6.8
See https://gist.github.com/3799ba28fce27923005300bf27d7504e for a full test report.

**Comment by bedroge** [comment]
@boegelbot please test @ jsc-zen2

**Review by bedroge — APPROVED** [review]
Lgtm

**Comment by bedroge** [comment]
Going in, thanks @sassy-crick!

**Pull request merged**